### PR TITLE
[DPE-5265] Add CI test pipeline for Hadoop

### PR DESCRIPTION
**Pull request by welpaolo**
This PR introduces workflow for running integration tests for Spark and Hadoop.

### CODE REVIEW

File: .github/workflows/test-hadoop.yaml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  cancel-in-progress: true
+
+on:
+  workflow_call:
+  workflow_dispatch:
+    inputs:
+      branches:
+        description: 'Comma-separated list of branches to test'
+        required: true
+        default: 'lp-3.3.6,lp-3.3.6,lp-3.3.6'
```
Comment (theoctober19th) [inline .github/workflows/test-hadoop.yaml:14]: question, why do we have all three entries the same?

**Review by theoctober19th — APPROVED** [review]
LGTM

File: ie-tests/tests.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+echo " "
+echo " Cleaning ..."
+mvn clean
+
+echo " "
+echo " Running tests..."
+# Run maven tests 
+mvn test -fn 
+# -Dhadoop.version=3.3.6-ubuntu1
```
Comment (deusebio) [inline ie-tests/tests.sh:18]: **niptick** this is a left over, we could probably remove this now

File: ie-tests/validate.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    args = parse_args()
+
+    zip_file_folder = args.archive_logs
+    print("Start analyze the logs from the different runs...")
+    logs_archives = os.listdir(zip_file_folder)
+    log_results = {}
+    log_errors = {}
+    log_tests = {}
+    print(f"Number of runs: {len(logs_archives)}")
+    for log_archive in logs_archives:
```
Comment (deusebio) [inline ie-tests/validate.py:50]: **suggestion** This for looks very ugly, and very nested. 

I would try to use generators to get something like:

```
@dataclass
class Result:
   run: int
   failures: int
   errors: int
   skipped: int
   raw: str

class Parser:
    # Parsing logic, we could have one for hadoop and one for spark
    @abstractmethod
    def parse(log_file:str) -> Result:
         ....

for log_file in extract_logs(zip_file_folder):
    result: Result = parser.parse(log_file)
    
    with open(f"/tmp/cleaned_{log_file}", "w") as f:
        f.write(result.raw)
```

File: ie-tests/validate.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    for file, results in log_results.items():
+        r = 0
+        f = 0
+        e = 0
+        s = 0
+        for res in results:
+            r += res[0]
+            f += res[1]
+            e += res[2]
+            s += res[3]
```
Comment (deusebio) [inline ie-tests/validate.py:114]: **suggestion** using tuple is a bit error prone and unreadable, I feel

File: ie-tests/validate_spark.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
+import os
+import re
+import tempfile
+import zipfile
+from argparse import ArgumentParser, Namespace
+
+from prettytable import PrettyTable
+
+
+def remove_ascii_colors(line):
```
Comment (deusebio) [inline ie-tests/validate_spark.py:10]: **suggestion** we could probably use the same function

**Review by deusebio — APPROVED** [review]
I'm fine to merge this straight away to unblock testing from Security, but I feel that the code base could be improved to be a bit to be more maintainable. 

Probably we could also have a general main and abstract the parsing, providing two different implementations one for Hadoop and one for Spark. I don't think we are very far from this, possibly just a few hours away, so maybe worth addressing in the next pulse or we will never do this. I provide some suggestion in the review

File: .github/workflows/test-hadoop.yaml (hunk 1 above)
Comment (welpaolo) [reply .github/workflows/test-hadoop.yaml:14]: It is an example to show that you can run test on a branch multiple times

**Pull request merged**